edit-post/components/plugin-more-menu-item/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { Slot, Fill, withContext, MenuItemsGroup, MenuItemsItem } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Name of slot in which the sidebar should fill.
+ *
+ * @type {String}
+ */
+const SLOT_NAME = 'PluginMoreMenuItem';
+
+function PluginMoreMenuItem( { label, onClick, icon, isSelected } ) {
+	return (
+		<Fill name={ SLOT_NAME }>
+			{ ( props ) => {
+				return (
+					<MenuItemsItem
+						icon={ isSelected ? 'yes' : icon }
+						isSelected={ isSelected }
+						label={ label }
+						onClick={ () => {
+							onClick();
+							props.onClose();
+						} }
+					/>
+				);
+			} }
+		</Fill>
+	);
+}
+
+PluginMoreMenuItem = compose( [
+	withContext( 'pluginName' )( ( pluginName, { target } ) => {
+		return {
+			target: `${ pluginName }/${ target }`,
+		};
+	} ),
+	withSelect( ( select, { target } ) => ( {
```

Comment:
Compactness ought not to be a goal. There's a lot of prior knowledge being assumed here with destructuring of `ownProps` and careful eyes required to discern the implicit object return.

Contrast with the verbose alternative, which I find to be much more readable:

```js
withSelect( ( select, ownProps ) => {
	const { target } = ownProps;
	const { getActiveGeneralSidebarName } = select( 'core/edit-post' );

	return {
		isSelected: getActiveGeneralSidebarName() === target,
	};
} ),
```

May even want a dedicated selector for this behavior, `isActiveGeneralSidebar( target )`